src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -85,10 +88,41 @@ public Response forward(@NotNull @Valid RestAction ra) {
         receivedActions.increment();
 
         try {
-            restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
-        } catch (BadRequestException e) {
-            // The following line is required to forward the HTTP 400 error message.
-            return Response.status(BAD_REQUEST).entity(e.getMessage()).build();
+            final Response response = this.restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
+
+            // Close the response to avoid memory leaks.
+            if (response != null) {
+                response.close();
+            }
+        } catch (final WebApplicationException e) {
+            // Build a nice error message for the caller.
+            final Response response = e.getResponse();
+            final String incomingErrorMessage = response.readEntity(String.class);
+            // Close the response to avoid memory leaks.
+            response.close();
+
+            Log.debugf(
```

Comment:
I wonder if we should log at `warn` or `error` everything except 400.

Reply:
Changed!